### DISCUSSION TITLE
Is a BufferedStream wrapper around files useful?

### DISCUSSION THREAD

**Discussion by cremor** [UglyToad/PdfPig#413] [Q&A]
Hi @EliotJones,

In https://github.com/UglyToad/PdfPig/issues/47#issuecomment-522709042 you recommended to use a `BufferedStream` as a wrapper around a `NetworkStream` when reading PDFs. Do you still recommend that? Or have there been any changes to PdfPig since that comment so that this isn't needed any more?

In https://github.com/UglyToad/PdfPig/pull/253#discussion_r554089632 you also recommend a `BufferedStream` around a `FileStream`. Did you test that this actually improves performance?
I'm asking because I found this in the .NET source code:

```cs
BCLDebug.Perf(!(stream is FileStream), "FileStream is buffered - don't wrap it in a BufferedStream");
```
Source: https://referencesource.microsoft.com/#mscorlib/system/io/bufferedstream.cs,100

Also, what about `PdfDocumentBuilder`? Does it also do random seeking/writing?

**Comment by EliotJones** [comment]
Hi @cremor I've never actually tested either approach I always just worked with the bytes directly. It makes sense then not to use it for filestream. As for networkstream I've not worked with one in the past, I'd be surprised it's even supported since I assume it wouldn't support seek operations but if it is then consumers would need to profile to determine if bufferedstream helps.

**Reply by cremor** [reply]
Thanks. I might try to run some benchmarks with the test PDFs from @vadik299 in https://github.com/UglyToad/PdfPig/issues/47#issuecomment-525992462 if I find some time. Do you maybe have other sample PDFs that could be useful in a benchmark?

**Reply by EliotJones** [reply]
This archive has over 1k files of various sizes and types https://github.com/UglyToad/PdfPig/discussions/302
